experimental/packages/opentelemetry-instrumentation/src/instrumentation.ts

```diff
@@ -173,4 +175,29 @@ export abstract class InstrumentationAbstract implements Instrumentation {
     | InstrumentationModuleDefinition
     | InstrumentationModuleDefinition[]
     | void;
+
+  /**
+   * Execute instrumentation event hook, if configured, and log any errors.
+   * Any semantics of the event type and eventInfo are defined by the specific instrumentation.
+   * @param hookHandler The optional hook handler which the user has configured via instrumentation config\
+   * @param eventName The name of the event being triggered, for logging purposes
+   * @param span The span to which the hook should be applied
+   * @param eventInfo The event info to be passed to the hook
+   */
+  protected runInstrumentationEventHook<EventInfoType>(
+    hookHandler: InstrumentationEventHook<EventInfoType> | undefined,
+    eventName: string,
+    span: Span,
+    eventInfo: EventInfoType
+  ) {
+    if (!hookHandler) {
+      return;
+    }
+
+    try {
+      hookHandler(span, eventInfo);
```

Comment:
I was wondering if it can be useful to also always include the `eventName` as an attribute in the event info, something like:
```suggestion
      hookHandler(span, {eventName, ...eventInfo});
```

and adjust the types.

The hook can always choose to ignore this parameter but it will be available if needed. Not sure thought if it's worth adding more complications to the code for a non existing use.

Reply:
This would be helpful if a hook function is used for several hooks. I have no usecase in mind. Price is the extra object created.

Reply:
My thinking as well. thanks. will wait for more opinions.

We can also add it in the future in a non breaking manner